components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -173,29 +222,81 @@ class Nimbus(
         nimbus.getExperimentBranch(experimentId)
 
     override fun updateExperiments() {
-        scope.launch {
-            try {
-                nimbus.updateExperiments()
-
-                // Get the experiments to record in telemetry
-                nimbus.getActiveExperiments().let {
-                    if (it.any()) {
-                        recordExperimentTelemetry(it)
-                        // The current plan is to have the nimbus-sdk updateExperiments() function
-                        // return a diff of the experiments that have been received, at which point we
-                        // can emit the appropriate telemetry events and notify observers of just the
-                        // diff
-                        notifyObservers { onUpdatesApplied(it) }
-                    }
-                }
-            } catch (e: ErrorException.RequestError) {
-                logger.info("Error fetching experiments from endpoint: $e")
-            } catch (e: ErrorException.InvalidExperimentResponse) {
-                logger.info("Invalid experiment response: $e")
+        networkScope.launch {
+            fetchExperimentsOnThisThread()
+            applyPendingExperimentsOnThisThread()
+        }
+    }
+
+    override fun fetchExperiments() {
+        networkScope.launch {
+            fetchExperimentsOnThisThread()
+        }
+    }
+
+    @WorkerThread
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    private fun fetchExperimentsOnThisThread() {
+        try {
+            nimbus.fetchExperiments()
+            notifyObservers { onExperimentsFetched() }
+        } catch (e: ErrorException.RequestError) {
+            logger.info("Error fetching experiments from endpoint: $e")
+        }
+    }
+
+    override fun applyPendingExperiments() {
+        dbScope.launch {
```

Comment:
Since this is now running in a separate coroutine scope, is it possible that this could still potentially block in the nimbus-sdk on the database since both `fetchExperiments` and `applyPendingExperiment` open a database writer and since both of those functions now operate in different scopes, whichever opened the writer first would block the other scope until it was finished.

Reply:
Right, I wasn't sure that it would result in any harm after looking at it, this was more of a "here be dragons" sort of question knowing that there was synchronization around the rkv database writer.